### DISCUSSION TITLE
react-select default value selected but get validation error

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5544] [Q&A] [answered]
Posted this twice by accident. Don't know how to remove the other one!

I am populating my react-select dropdown with a default value from the server which displays correctly but when I try to submit the form the validation error shows up so it isn't actually picking up the value. How can I fix this? 

I am using **V7** of react-hook-form

```
  const defaultValues = {
    customerId: invoice.customerId._id,
    customerLabel:
      invoice.customerId.companyName +
      " " +
      invoice.customerId.firstName +
      " " +
      invoice.customerId.lastName,
        items: [{}],
  };

  const {
    register,
    control,
    handleSubmit,
    formState: { errors },
    clearErrors,
  } = useForm({
    mode: "onChange",
    defaultValues,
    resolver: yupResolver(schema),
  });


<Controller
  name="customer"
  control={control}
  render={({ field }) => (
    <Select
      {...field}
      options={options}
      defaultValue={{
        label: defaultValues.customerLabel,
        value: defaultValues.customerId,
      }}
    />
  )}
/>
```

**Comment by bluebill1049** [comment]
CSB

**Reply by ghost** [reply]
You know what, I just realised that I am being an absolute tool. I figured out my problem so you can please go ahead and close this as well as my duplicate. Apologies for all the trouble!

**Reply by bluebill1049** [reply]
no worries. would like to post the solution to the problem. we can mark it as answer so it help others too.

**Comment by ghost** [comment, marked as answer]
I am not really sure if this solved the issue but it is working now. I played around so much I can't remember everything I did. But one thing is that I put the variables directly into defaultValue instead of using from defaultValues const

```
  <Controller
    name="customer"
    defaultValue={{
      value:  invoice.customerId._id,
      label:  invoice.customerId.firstName +  " " +  invoice.customerId.lastName
    }}
    render={({ field }) => <Select {...field} options={options} />}
    control={control}
  />
```
